crates/matrix-sdk-ui/src/timeline/traits.rs

```diff
@@ -160,128 +164,110 @@ impl RoomDataProvider for Room {
-                .load_event_receipts(ReceiptType::Read, ReceiptThread::Unthreaded, event_id)
-                .await
-            {
-                Ok(receipts) => receipts.into_iter().collect(),
-                Err(e) => {
-                    error!(?event_id, "Failed to get unthreaded read receipts for event: {e}");
-                    IndexMap::new()
-                }
-            };
+    ) -> IndexMap<OwnedUserId, Receipt> {
+        let mut unthreaded_receipts = match self
+            .load_event_receipts(ReceiptType::Read, ReceiptThread::Unthreaded, event_id)
+            .await
+        {
+            Ok(receipts) => receipts.into_iter().collect(),
+            Err(e) => {
+                error!(?event_id, "Failed to get unthreaded read receipts for event: {e}");
+                IndexMap::new()
+            }
+        };
 
-            let main_thread_receipts = match self
-                .load_event_receipts(ReceiptType::Read, ReceiptThread::Main, event_id)
-                .await
-            {
-                Ok(receipts) => receipts,
-                Err(e) => {
-                    error!(?event_id, "Failed to get main thread read receipts for event: {e}");
-                    Vec::new()
-                }
-            };
+        let main_thread_receipts = match self
+            .load_event_receipts(ReceiptType::Read, ReceiptThread::Main, event_id)
+            .await
+        {
+            Ok(receipts) => receipts,
+            Err(e) => {
+                error!(?event_id, "Failed to get main thread read receipts for event: {e}");
+                Vec::new()
+            }
+        };
 
-            unthreaded_receipts.extend(main_thread_receipts);
-            unthreaded_receipts
-        }
-        .boxed()
+        unthreaded_receipts.extend(main_thread_receipts);
+        unthreaded_receipts
     }
 
-    fn push_rules_and_context(&self) -> BoxFuture<'_, Option<(Ruleset, PushConditionRoomCtx)>> {
-        async {
-            match self.push_context().await {
-                Ok(Some(push_context)) => match self.client().account().push_rules().await {
-                    Ok(push_rules) => Some((push_rules, push_context)),
-                    Err(e) => {
-                        error!("Could not get push rules: {e}");
-                        None
-                    }
-                },
-                Ok(None) => {
-                    debug!("Could not aggregate push context");
-                    None
-                }
+    async fn push_rules_and_context(&self) -> Option<(Ruleset, PushConditionRoomCtx)> {
+        match self.push_context().await {
+            Ok(Some(push_context)) => match self.client().account().push_rules().await {
+                Ok(push_rules) => Some((push_rules, push_context)),
                 Err(e) => {
-                    error!("Could not get push context: {e}");
+                    error!("Could not get push rules: {e}");
```

Comment:
:+1:

Reply:
Err, this is not something I changed..

Reply:
Checked the diff, it's all good. This is just git displaying the change in a very confusing way because indentation changed. This was already correct before.

Reply:
Oh yeah right, sorry for the noise.